code/game/objects/items/cigs_lighters.dm

```diff
@@ -212,7 +212,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 				return
 		reagents.remove_any(REAGENTS_METABOLISM)
 
-
+var/counter = 0 // There is no way this is the best way of doing this

```

Comment:
Pretty sure this goes in /obj/item/clothing/mask/cigarette, dummy

Put it under var/smoketime

I'd also rename it to something a bit more descriptive, like "nextdragtime"

Reply:
ye sure ok